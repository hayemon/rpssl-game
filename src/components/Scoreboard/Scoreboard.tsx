import React from 'react'
import { RiDeleteBack2Fill } from 'react-icons/ri'
import { useDispatch } from 'react-redux'
import { clearResultsHistory } from 'store/actions'
import { selectResultsHistory } from 'store/selectors'
import styles from './index.module.scss'

const Scoreboard: React.FC = () => {
  const dispatch = useDispatch()
  const resultsHistory = selectResultsHistory()

  const handleClick = () => {
    dispatch(clearResultsHistory())
  }

  const hasHistory = resultsHistory && resultsHistory.length > 0

  return (
    <div className={`glass ${styles.box}`}>
      <div className={styles.title}>
        <h3 data-testid='title'>Scoreboard</h3>
        <RiDeleteBack2Fill onClick={handleClick} data-testid='icon' />
      </div>
      <hr data-testid='divider' />
      {hasHistory && (
        <ul data-testid='list'>
          {resultsHistory.map((results) => (
            <li key={results.timeStamp}>{results.results}</li>
          ))}
        </ul>
      )}
    </div>
  )
}

export default Scoreboard
