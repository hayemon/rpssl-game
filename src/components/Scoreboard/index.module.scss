@import '../../variables.scss';

.box {
  position: absolute;
  top: 0;
  right: 0;
  width: 160px;
  height: auto;
  border-radius: $border-radius;
  margin: $margin;
  padding: $padding;
}

.title {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.title h3 {
  margin: 0;
  padding: 0;
  letter-spacing: 3px;
  font-weight: lighter;
}

.title svg {
  font-size: 1.5rem;
  overflow: visible;
}

.title svg:hover {
  filter: drop-shadow(0px 0px 1px white);
}

.title svg:active {
  transform: scale(1.05);
}

.box hr {
  border: 1px solid rgba(255, 255, 255, 0.5);
}

.box ul {
  width: 100%;
  margin: 0;
  padding: 0;
}

.box ul li {
  display: flex;
  margin: $margin 0;
  transition: 0.5s;
  letter-spacing: 2px;
}

.box ul li:hover {
  transform: translateX(10px) scale(1.1);
}
