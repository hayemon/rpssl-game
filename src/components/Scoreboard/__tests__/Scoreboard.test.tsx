import '@testing-library/jest-dom/extend-expect'
import { fireEvent } from '@testing-library/react'
import { renderWithProviders } from 'store/test/testUtils'
import Scoreboard from '../Scoreboard'

describe('Scoreboard', () => {
  it('renders', () => {
    const { asFragment } = renderWithProviders(<Scoreboard />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('check elements', async () => {
    const { getByTestId } = renderWithProviders(<Scoreboard />)

    expect(getByTestId('title')).toHaveTextContent('Scoreboard')
    expect(getByTestId('icon')).toBeInTheDocument()
    expect(getByTestId('divider')).toBeInTheDocument()
    expect(getByTestId('list')).toBeInTheDocument()
  })

  it('clear history', async () => {
    const { getByTestId, queryByTestId } = renderWithProviders(<Scoreboard />)

    fireEvent.click(getByTestId('icon'))
    expect(queryByTestId('list')).toBeNull()
  })
})
