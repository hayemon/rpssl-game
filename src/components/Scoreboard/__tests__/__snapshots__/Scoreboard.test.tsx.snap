// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard renders 1`] = `
<DocumentFragment>
  <div
    class="glass undefined"
  >
    <div>
      <h3
        data-testid="title"
      >
        Scoreboard
      </h3>
      <svg
        data-testid="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M6.535 3H21a1 1 0 0 1 1 1v16a1 1 0 0 1-1 1H6.535a1 1 0 0 1-.832-.445l-5.333-8a1 1 0 0 1 0-1.11l5.333-8A1 1 0 0 1 6.535 3zM13 10.586l-2.828-2.829-1.415 1.415L11.586 12l-2.829 2.828 1.415 1.415L13 13.414l2.828 2.829 1.415-1.415L14.414 12l2.829-2.828-1.415-1.415L13 10.586z"
          />
        </g>
      </svg>
    </div>
    <hr
      data-testid="divider"
    />
    <ul
      data-testid="list"
    >
      <li>
        win
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
